destination.go

```diff
@@ -79,17 +79,22 @@ func (d *Destination) Open(ctx context.Context) (err error) {
 }
 
 func (d *Destination) Write(ctx context.Context, records []sdk.Record) (int, error) {
-	for i, record := range records {
-		err := sdk.Util.Destination.Route(ctx, record,
-			d.writer.Upsert,
-			d.writer.Upsert,
-			d.writer.Delete,
-			d.writer.Upsert,
-		)
+	upsertVecs, deleteIds, written, err := parseRecords(ctx, records)
```

Comment:
So, IIUC, here we collect all records to be upserted in one batch and the same for records to be deleted. However, those records might be "mixed" in the input `records` and that could complicate the logic.

For example, let's say we have 2 upserts, 2 deletes, and then again 2 upserts. That would mean one batch with 4 upserts and then one with 2 deletes. Now, let's say the second delete fails. We need to do: `return 3 , err`.

That can be solved, but there's another thing: we've upserted 2 (or more) records for nothing. So it's probably better not to build 2 batches (one for upserts and one for deletes), but to build batches as they come up in the input `records`. In our case that would be three batches: 2 upserts, 2 deletes, 2 upserts.

Reply:
yeah! that makes sense. Was worrying basically about that on slack.

I'll update the logic.